compiler/z/codegen/ControlFlowEvaluator.cpp

```diff
@@ -2724,99 +2724,50 @@ TR::Register *
-
-      cg->stopUsingRegister(firstCmpReg);
-      cg->stopUsingRegister(secondCmpReg);
-      }
-   else
+   TR::Register *resultReg = cg->gprClobberEvaluate(trueValueNode);
+   TR::Register *conditionReg = cg->evaluate(conditionNode);
+   TR::Register *falseValReg = cg->evaluate(falseValueNode);
+   if (cg->comp()->target().cpu.getSupportsArch(TR::CPU::z13) && node->getOpCode().isDouble())
       {
-      TR::Register *conditionReg = cg->evaluate(conditionNode);
+      TR::Register *vectorSelReg = cg->allocateRegister(TR_VRF);
       TR::Register *tempReg = cg->allocateRegister(TR_FPR);
       TR::Register *vzeroReg = cg->allocateRegister(TR_VRF);
-
+      // Convert 32 Bit register to 64 Bit (Comparison Child of the select node is 32 bit)
+      generateRRInstruction(cg, TR::InstOpCode::LLGFR, node, conditionReg, conditionReg);
       // convert to floating point
       generateRRInstruction(cg, TR::InstOpCode::LDGR, node, tempReg, conditionReg);
-
       // generate compare with zero
       generateVRIaInstruction(cg, TR::InstOpCode::VGBM, node, vzeroReg, 0, 0);
       generateVRRcInstruction(cg, TR::InstOpCode::VFCE, node, vectorSelReg, tempReg, vzeroReg, 1, 0, 3);
-
       // generate select - if condition == 0, vectorSelReg will contain all 1s, so false and true are swapped
-      generateVRReInstruction(cg, TR::InstOpCode::VSEL, node, returnReg, falseValueReg, trueValueReg, vectorSelReg);
-
+      generateVRReInstruction(cg, TR::InstOpCode::VSEL, node, resultReg, falseValReg, resultReg, vectorSelReg);
       cg->stopUsingRegister(tempReg);
       cg->stopUsingRegister(vzeroReg);
-      cg->stopUsingRegister(conditionReg);
+      cg->stopUsingRegister(vectorSelReg);
       }
-
-   node->setRegister(returnReg);
-
-   cg->stopUsingRegister(trueValueReg);
-   cg->stopUsingRegister(falseValueReg);
-   cg->stopUsingRegister(vectorSelReg);
-
+   else
+      {
+      TR::LabelSymbol *doneLabel = generateLabelSymbol(cg);
+      TR::LabelSymbol *startICF = generateLabelSymbol(cg);
+      generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node , startICF);
+      TR::RegisterDependencyConditions* conditions = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 3, cg);
+      conditions->addPostCondition(resultReg, TR::RealRegister::AssignAny);
+      conditions->addPostCondition(falseValReg, TR::RealRegister::AssignAny);
+      conditions->addPostCondition(conditionReg, TR::RealRegister::AssignAny);
+      generateS390CompareAndBranchInstruction(cg, TR::InstOpCode::CL, node, conditionReg, 0, TR::InstOpCode::COND_BNE, doneLabel, false, false);
+      generateRRInstruction(cg, node->getOpCode().isDouble() ? TR::InstOpCode::LDR : TR::InstOpCode::LER, node,  resultReg, falseValReg);
+      generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node, doneLabel, conditions);
+      startICF->setStartInternalControlFlow();
+      doneLabel->setEndInternalControlFlow();
```

Comment:
I think in 72fbbb2814fb9901b825599b976de727180d604b we've canonicalized the naming scheme to be `cFlowRegionStart` / `cFlowRegionEnd` for consistency. Could we make the change here as well?

Reply:
Changed it in https://github.com/eclipse/omr/pull/4996/commits/1061c0c6d0fb96e7c8988f2af157f19ef9ae104c